service/auth.ts

```diff
@@ -91,11 +92,15 @@ export async function createUser(
   email: string,
   password: string
 ) {
-  return client.post<CreateUserReq, CreateUserRes>("users/", {
-    username,
-    email,
-    password,
-  });
+  return http.post<CreateUserReq, CreateUserRes>(
```

Comment:
```suggestion
  return http.post(
```
Minor thing: but in places in this file where you are explicitly passing in the types in the generic params, you should be able to omit them as TypeScript will automatically infer the type of them based on what you passed in, so less code to write that way!

Reply:
Ahh yeah I think that's because the `response.json(...)` bit from the fetch response can literally be anything, so we'd need to specify it 👍